packages/appsync-modelgen-plugin/src/utils/process-index.ts

```diff
@@ -9,21 +9,27 @@ import { toLower, toUpper } from './stringUtils';
  */
 export const processIndex = (model: CodeGenModel) => {
   const indexMap = model.fields.reduce((acc, field) => {
-    const indexDirective = getDirective(field)('index');
-    if (!indexDirective) {
+    const indexDirectives = getDirectives(field)('index');
+    if (!indexDirectives) {
       return acc;
     }
-    return { ...acc, [field.name]: indexDirective };
-  }, {} as Record<string, CodeGenDirective>);
+    return { ...acc, [field.name]: indexDirectives };
+  }, {} as Record<string, CodeGenDirective[]>);
+
+  const keyList: CodeGenDirective[] = [];
+  Object.entries(indexMap).forEach(([fieldName, directives]) => {
```

Comment:
nit: Why change this from a map to a forEach?

Reply:
`.map` returns a new array, whereas `.forEach` doesn't return anything. Since the refactored code doesn't rely on the returned array, `.forEach` is a better fit here.